src/ts/types/__tests__/management.ts

```diff
@@ -32,7 +33,7 @@ export const management = async () => {
     });
 
     TrezorConnect.firmwareUpdate({
-        payload: new ArrayBuffer(0),
+        binary: [],
```

Comment:
```suggestion
        binary: Buffer.from('abcd'),
```
ts should scream here, if you use array, no?

Reply:
Actually it does not, quite the opposite, applying your suggestion results in error
```
type 'Buffer' is missing the following properties from type 'number[]': pop, push, concat, shift, and 3 more.
36         binary: Buffer.from('abcd'),
           ~~~~~~
```

the `TrezorConnect.firmwareUpdate` [is typed](https://github.com/trezor/connect/blob/e8efb21fe44d752b3e76a151e7eb661a46502087/src/ts/types/api.d.ts#L347-L349) as
```typescript
function firmwareUpdate(
        params: P.CommonParams & (Mgmnt.FirmwareUpdate | Mgmnt.FirmwareUpdateBinary),
    ): P.Response<P.DefaultMessage>;
```

and the `FirmwareUpdate` and `FirmwareUpdateBinary` are defined [here](https://github.com/trezor/connect/blob/e8efb21fe44d752b3e76a151e7eb661a46502087/src/ts/types/trezor/management.d.ts#L31-L39) and indeed the `binary` field is of type `number[]`.

I didn't consider that as wrong and so I simply alligned the type here in the "tests" rather than in the API which would be a breaking change. But maybe I'm missing something...?

Reply:
actually this is a bug in ts definitions.
as you noticed ts is written by hand and never tested before (and it's not used in compile time) 
flowtype and protobuf definitions are the source of truth here:
https://github.com/trezor/connect/blob/develop/src/js/types/trezor/protobuf.js#L567
https://github.com/trezor/connect/blob/develop/src/js/types/trezor/management.js#L31
https://github.com/trezor/connect/blob/develop/src/js/core/methods/FirmwareUpdate.js#L38

this `FirmwareUpdateBinary` was prepared for a feature called "upload custom firmware" is trezor-suite and never implemented (same as safety_checks) this is the reason for not catching this bug before.

don't worry. this is not a breaking change, it's a bug fix

Reply:
Aaah, ok. I changed the type to `Buffer` in https://github.com/trezor/connect/pull/785/commits/f513e33a7094ef1d6a400af511e732e3857130ee

I also added a record to changelog.